examples/sparse/hetero-rgcn.py

```diff
@@ -0,0 +1,323 @@
+    labels = g.nodes[category].data.pop("labels")
+
+    # Split dataset into train, validate, test.
+    if args.validation:
+        val_idx = train_idx[: len(train_idx) // 5]
+        train_idx = train_idx[len(train_idx) // 5 :]
+    else:
+        val_idx = train_idx
+
+    # Check cuda.
+    use_cuda = args.gpu >= 0 and th.cuda.is_available()
+    if use_cuda:
+        th.cuda.set_device(args.gpu)
+        g = g.to("cuda:%d" % args.gpu)
+        labels = labels.cuda()
+        train_idx = train_idx.cuda()
+        test_idx = test_idx.cuda()
+
+    # Create model.
+    model = EntityClassify(
+        args.n_hidden,
+        num_classes,
+        list(set(g.etypes)),
+    )
+
+    embed_layer = RelGraphEmbed(
+        {ntype: g.num_nodes(ntype) for ntype in g.ntypes}, args.n_hidden
+    )
+
+    if use_cuda:
+        model.cuda()
+
+    # Optimizer.
+    optimizer = th.optim.Adam(
+        model.parameters(), lr=args.lr, weight_decay=args.l2norm
+    )
+
+    # Construct hetero sparse matrix.
+    A = {}
+    for stype, etype, dtype in g.canonical_etypes:
+        eg = g[stype, etype, dtype]
+        indices = th.stack(eg.edges("uv"))
+        A[(stype, etype, dtype)] = dglsp.spmatrix(
+            indices, shape=(g.num_nodes(stype), g.num_nodes(dtype))
+        )
+
+        D1_hat = dglsp.diag(A[(stype, etype, dtype)].sum(1)) ** -0.5
+        D2_hat = dglsp.diag(A[(stype, etype, dtype)].sum(0)) ** -0.5
+        A[(stype, etype, dtype)] = D1_hat @ A[(stype, etype, dtype)] @ D2_hat
+
+    # Training loop.
+    print("start training...")
+    model.train()
+    for epoch in range(args.n_epochs):
+        optimizer.zero_grad()
+        logits = model(embed_layer(), A)[category]
+        loss = F.cross_entropy(logits[train_idx], labels[train_idx])
+        loss.backward()
+        optimizer.step()
+
+        train_acc = th.sum(
+            logits[train_idx].argmax(dim=1) == labels[train_idx]
+        ).item() / len(train_idx)
+        val_loss = F.cross_entropy(logits[val_idx], labels[val_idx])
+        val_acc = th.sum(
+            logits[val_idx].argmax(dim=1) == labels[val_idx]
+        ).item() / len(val_idx)
+        print(
+            f"Epoch {epoch:05d} | Train Acc: {train_acc:.4f} | "
+            f"Train Loss: {loss.item():.4f} | Valid Acc: {val_acc:.4f} | "
+            f"Valid loss: {val_loss.item():.4f} "
+        )
+    print()
+    if args.model_path is not None:
+        th.save(model.state_dict(), args.model_path)
+
+    model.eval()
+    logits = model.forward(embed_layer(), A)[category]
+    test_loss = F.cross_entropy(logits[test_idx], labels[test_idx])
+    test_acc = th.sum(
+        logits[test_idx].argmax(dim=1) == labels[test_idx]
+    ).item() / len(test_idx)
+    print(
+        "Test Acc: {:.4f} | Test loss: {:.4f}".format(
+            test_acc, test_loss.item()
+        )
+    )
+    print()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="RGCN")
+    parser.add_argument(
```

Comment:
To simplified this example, you only need to keep the option of datasets. In particular,`gpu`, `model_path`, `validation`, and `testing` are not needed. `n-hidded`, `lr`, `n-epochs`, `l2norm` can be defined when they are used.

Reply:
OK, most args have been removed.